Mathlib/Algebra/Order/Module/OrderedSMul.lean

```diff
@@ -0,0 +1,147 @@
+/-
+Copyright (c) 2020 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+import Mathlib.Algebra.Module.Pi
+import Mathlib.Algebra.Module.Prod
+import Mathlib.Algebra.Order.Module.Defs
+import Mathlib.Algebra.Order.Monoid.Prod
+import Mathlib.Algebra.Order.Pi
+import Mathlib.Data.Set.Pointwise.SMul
+import Mathlib.Tactic.GCongr.Core
+import Mathlib.Tactic.Positivity
+
+#align_import algebra.order.smul from "leanprover-community/mathlib"@"3ba15165bd6927679be7c22d6091a87337e3cd0c"
+
+/-!
+# Ordered scalar product
+
+In this file we define
+
+* `OrderedSMul R M` : an ordered additive commutative monoid `M` is an `OrderedSMul`
+  over an `OrderedSemiring` `R` if the scalar product respects the order relation on the
+  monoid and on the ring. There is a correspondence between this structure and convex cones,
+  which is proven in `Analysis/Convex/Cone.lean`.
+
+## Implementation notes
+* We choose to define `OrderedSMul` as a `Prop`-valued mixin, so that it can be
+  used for actions, modules, and algebras
+  (the axioms for an "ordered algebra" are exactly that the algebra is ordered as a module).
+* To get ordered modules and ordered vector spaces, it suffices to replace the
+  `OrderedAddCommMonoid` and the `OrderedSemiring` as desired.
+
+## TODO
+
+This file is now mostly useless. We should try deleting `OrderedSMul`
```

Comment:
What is it superseded by?

Reply:
The `PosSMulMono` and friends mixins. `OrderedSMul` is a mixin that effectively combines`PosSMulStrictMono` and `PosSMulReflectLT` together.